stdlib/public/core/Result.swift

```diff
@@ -0,0 +1,79 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@_frozen
+public enum Result<Value, Error> {
+    case success(Value)
+    case failure(Error)
+    
+    var value: Value? {
+        switch self {
+        case let .success(value): return value
+        case .failure: return nil
+        }
+    }
+    
+    var error: Error? {
+        switch self {
+        case let .failure(error): return error
+        case .success: return nil
+        }
+    }
+    
+    var isSuccess: Bool {
+        switch self {
+        case .success: return true
+        case .failure: return false
+        }
+    }
+    
+    @_transparent
+    public init(_ throwing: () throws -> Value) {
+        do {
+            let value = try throwing()
+            self = .success(value)
+        } catch {
+            self = .failure(error as! Error)
+        }
+    }
```

Comment:
test for this looks like it'd be nice; impl looks fine by inspection - but for such a simple type, no sense not aiming for  maximal coverage.